Fixed    #2039 Allows for configmaptocrs to modify address pools and BFD profiles if…

Sebastian-RG opened the pull request:
… they would not have valid resource names once converted

Thanks for sending a pull request! A few things before we get started:

1. If this is your first time, please read the [contributing guide](https://metallb.universe.tf/community/#contributing)
:heavy_check_mark: 
2. For non-trivial pull requests, please [file an
   issue](https://github.com/metallb/metallb/issues/new) first, and get
   agreement that the change is a good idea, and a general guideline
   for how it should be implemented, before sending code. Large PRs
   that weren't first discussed and agreed upon in an issue won't be
   accepted.
https://github.com/metallb/metallb/issues/2039
3. If the PR fixes a particular bug, please include the words "Fixed
   #<issue number>" in the PR text, so that the bug auto-closes when
   the PR is merged.
Fixed  #2039

Sebastian-RG commented:
@oribon Hi! If you have the time, could you approve the workflow so the automated tests can run?

oribon commented:
> @oribon Hi! If you have the time, could you approve the workflow so the automated tests can run?

sorry was away for a while, done and will review it soon I hope

Sebastian-RG commented:
Fixed an issue with how the `convertNames` flag was passed that was causing CI to fail. I'm not sure how to fix the [Implicit memory aliasing in for loop](https://github.com/metallb/metallb/actions/runs/5857332084/job/16059582956#step:4:162) one. I don't think I modified those lines.

Sebastian-RG commented:
Sorry about the mess :sweat:, I'm new to Golang and i was checking for an empty list/string the wrong way. This should work now

fedepaol commented:
> Sorry about the mess 😓, I'm new to Golang and i was checking for an empty list/string the wrong way. This should work now

no problem, and thanks a lot for the contribution! I will go over it soon

Sebastian-RG commented:
At least this time the actual unit tests passed :smiling_face_with_tear:. I'll correct what the linter mentions in a minute.

Sebastian-RG commented:
`gofmt -s` done

liornoy commented:
@Sebastian-RG we just merged a fix for the static-security-analysis, so can you please rebase this PR?
Also, in order to have an organized and unified git log in this repo we have guidelines for the commit messages format.
The full details can be found in here: https://cbea.ms/git-commit/
But if you don't want to read all of it, the basics are: 
Have a title up to 50 characters, starting with a capital letter,
separate the title from body with a blank line
and wrap the body at 72 characters

liornoy commented:
@fedepaol I think it's better to have this conversion happen by default. wdyt?

fedepaol commented:
> @fedepaol I think it's better to have this conversion happen by default. wdyt?

+1 , the old version is useless is the generated name is not valid.

liornoy commented on configmaptocrs/main.go:
```diff
@@ -149,6 +156,65 @@ func decodeConfigFile(raw []byte) (*configFile, error) {
+	var err error
+
+	for i := 0; i < len(cf.Pools); i++ {
+		cf.Pools[i].Name, err = formatToK8S(cf.Pools[i].Name, "IpAddressPool")
```
Can we maybe do a simple one-liner regex check on the name to see if it's invalid format before calling it straight away?
i.e. if it's valid continue to next iteration, else it will invoke the formatToK8S

liornoy commented:
@Sebastian-RG thanks for the fix, looks good, just left one comment from my side

Sebastian-RG commented:
Regex added as requested

liornoy commented:
lgtm, thanks!

fedepaol commented:
sent to merge queue

Sebastian-RG commented:
Fixed a golangci-lint error, sorry for not checking before pushing
